### ISSUE TITLE
How to include an entire directory?

### ISSUE THREAD

**Issue by Yuriusu** [termux/termux-create-package#20]
### EDIT

I found the line for the directory recursion:

```
data_tarfile.add(input_file_path, arcname=output_file, recursive=False)
```

> Note: Setting `recursive=False` to `True` will enable directory recursion.
> From Python docs: [tarfile.TarFile.add](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.add)

I think recursion should be enabled by default.

<hr />

I can't figure out how to include a directory. Does packaging an entire directory supported?

Suppose I have this setup:

```
.
├── bin
│   └── script
└── manifest.json
```

Can we just include the directory `bin` and be placed directly to the `prefix` (similar to how [package.json#files](https://docs.npmjs.com/files/package.json#files) works in [npm](https://npmjs.com))? Something like this:

```
"files": {
  "bin": "."
}
```

Would somehow result to this:

```
/data/data/com.termux/files/usr/bin/script
```

It would be very helpful for large number of files.

**Comment by xeffyr** [comment]
No, directory inclusion is not supported.

And generally, in my opinion, this script should be completely rewritten to use `dpkg-deb` for creating debfiles. This would fix issues like https://github.com/termux/termux-create-package/issues/18 and this one as packaging will be offloaded to standard utility for making debfiles.

**Comment by Yuriusu** [comment]
> No, directory inclusion is not supported.

As a workaround to my problem, I've edited the Python script and enabled recursion to true (see my post above).

> And generally, in my opinion, this script should be completely rewritten to use `dpkg-deb` for creating debfiles. This would fix issues like #18 and this one as packaging will be offloaded to standard utility for making debfiles.

I agree. It's been a couple of years since this is updated. It's time to rewrite this :smile: .

**Comment by Yuriusu** [comment]
Closing as fix merged with pull [#21](https://github.com/termux/termux-create-package/pull/21).
